python/docs/lis/userguide.rst

```diff
@@ -56,26 +65,180 @@ also be reached directly from the Logical File:
-   return empty arrays for these.
+    Please refer to :class:`dlisio.lis.DataFormatSpec` and
+    :class:`dlisio.core.spec_block_0` / :class:`dlisio.core.spec_block_1` for
+    the complete documentation of DFSR and Spec Blocks, respectively.
 
+Reading the curves
+..................
+
+The :class:`dlisio.lis.DataFormatSpec` only contains the metadata about the
+logset. To read the actual curves, pass the DFSR ``format_spec`` to
+:func:`dlisio.lis.curves` together with the filehandle ``f``:
+
+.. code-block:: python
+
+    >>> curves = lis.curves(f, format_spec)
+
+This returns a `structured numpy.ndarray`_ containing all the curves in the
+logset, using the mnemonics from the Spec Blocks as field (column) names.
+
+Fast Channels
+.............
+
+There is one caveat to reading the curves from a logset that is necessary to be
+aware of. LIS supports a concept it refers to as 'Fast Channels'. In short this
+means that channels within the *same* logset can have different sampling rates.
+
+When a logset contains curves with different sampling rates, these have to be
+read separately. I.e. multiple calls to the :func:`dlisio.lis.curves` is
+necessary in order to read all the curves. The sampling rate of a channel is
+recorded in its Spec Block:
+
+.. code-block:: python
+
+    >>> ch = format_spec.specs[-1]
+    >>> ch.mnem
+    'RCCL'
+    >>> ch.samples
+    6
+
+The last channel in this Data Format Spec, ``RCCL``, has a sampling rate of 6,
+which means it's sampled 6 times as frequent as the index channel. Remember,
+that the index ``DEPT`` was sampled once every 6th inch. That implicitly means
+that ``RCCL`` is sampled once every inch.
+
+You can also get a set of all sample rates in the DataFormatSpec:
+
+.. code-block:: python
+
+    >>> format_spec.sample_rates()
+    {1, 6, 30}
+
+There are 3 different sampling rates, which be must read with separate calls to
+:func:`dlisio.lis.curves()`:
+
+.. code-block:: python
+
+    >>> data_01 = lis.curves(f, format_spec, sample_rate=1)
+    >>> data_06 = lis.curves(f, format_spec, sample_rate=6)
+    >>> data_30 = lis.curves(f, format_spec, sample_rate=30)
+
+.. note::
+    Please refer to :func:`dlisio.lis.curves` for a more comprehensive
+    explanation of this topic.
+
+.. note::
+    When reading channels with a higher sampling rate, the index is still
+    included in the resulting numpy array, and the missing values have been
+    linearly interpolated between the samples in the file. This behaviour is
+    defined by LIS79 itself.
+
+Associate curves and metadata
+.............................
+
+:func:`dlisio.lis.curves` has a sister function
+:func:`dlisio.lis.curves_metadata`. This function can be useful if you want to
+associate the curves and metadata (Spec Blocks):
+
+.. code-block:: python
+
+    >>> data = lis.curves(f, format_spec, sample_rate=30)
+    >>> meta = lis.curves_metadata(format_spec, sample_rate=30)
```

Comment:
Maybe explicitly note that this doesn't need "f" reference? By other similarities it looks like you missed it here.

Reply:
Neh